add argmin, add_ ops

billmguo opened the pull request:
Summary: add argmin, add_ ops  annontation

Differential Revision: D68811662

billmguo commented:
@pytorchbot label "topic: not user facing"

billmguo commented:
@pytorchbot label "topic: not user facing"

cccclai commented on backends/qualcomm/builders/op_add.py:
```diff
@@ -15,7 +15,7 @@
 @register_node_visitor
 class Add(NodeVisitor):
-    target = ["aten.add.Tensor"]
+    target = ["aten.add.Tensor", "aten.add_.Tensor"]
```
I feel like `add_` technically is an in-place op, and qualcomm's add operator is not the same

cccclai commented on backends/qualcomm/tests/test_qnn_delegate.py:
```diff
@@ -1595,6 +1595,12 @@ def test_qnn_backend_chunk_add(self):
         module = self.get_qdq_module(module, sample_input)
         self.lower_module_and_test_output(module, sample_input)
+    
+     def test_qnn_backend_argmin(self):
```
Have you run this test?

billmguo replied:
the test passed with your test framework enabled

shewu-quic commented on backends/qualcomm/quantizer/annotators.py:
```diff
@@ -477,6 +479,11 @@ def annotate_log_softmax(node: Node, quantization_config: QuantizationConfig) ->
 
+@register_annotator([torch.ops.aten.argmin.default])
+def annotate_argmin(node: Node, quantization_config: QuantizationConfig) -> None:
+    annotate_single_in_single_out(node, quantization_config)
```
I think we don't need to insert QDQ for the output of the argmin node.

billmguo replied:
fixed it

shewu-quic commented on backends/qualcomm/tests/test_qnn_delegate.py:
```diff
@@ -1586,6 +1591,12 @@ def test_qnn_backend_view(self):
         module = self.get_qdq_module(module, sample_input)
         self.lower_module_and_test_output(module, sample_input)
 
+    def test_qnn_backend_argmin(self):
```
Could you make the order alphabetically? thanks.

shewu-quic commented on backends/qualcomm/tests/test_qnn_delegate.py:
```diff
@@ -681,6 +681,11 @@ def test_qnn_backend_view(self):
         sample_input = (torch.randn([1, 8, 512]), torch.randn([1, 2, 8, 256]))
         self.lower_module_and_test_output(module, sample_input)
 
+    def test_qnn_backend_argmin(self):
```
Could you make the order alphabetically? thanks.

shewu-quic reviewed (commented):
Thank you for your efforts.

shewu-quic commented on backends/qualcomm/builders/op_argmin.py:
```diff
@@ -0,0 +1,80 @@
+        )
+        argmin_input_tensors = [argmin_inp_tensor_wrapper]
+
+        output_tensor = self.get_tensor(node, node)
```
I think you are missing `.to(torch.int32)` here.
QNN is not support int64 output for argmin.
https://github.com/pytorch/executorch/blob/0a936e0e1ddf257f596de94af159e8f5d59ef26c/backends/qualcomm/builders/op_topk.py#L60

billmguo committed:
add argmin support (#8040)

Summary:

https://github.com/pytorch/executorch/pull/8040
add argmin, add_ ops  annontation

Differential Revision: D68811662

shewu-quic reviewed (approved):
LGTM

cccclai reviewed (approved):
Thank you!
